### PR TITLE
Chef 15 fixes

### PR DESCRIPTION
**Pull request by ramereth**
This imports the latest version of the code_generator cookbook from chef-cli and
updates it to OSL standards.

NOTE: This will break on ChefDK and requires Chef/Cinc Workstation

### CODE REVIEW

**Comment by ramereth** [comment]
[Update code_generator](https://trello.com/c/fX7uUabZ/1666-update-codegenerator)

File: files/default/build_cookbook/kitchen.yml [hunk 1]
```diff
@@ -11,8 +11,8 @@ provisioner:
   product_name: chefdk
```
Comment (detjensrobert) [inline files/default/build_cookbook/kitchen.yml:11]: Should the default `kitchen.yml` have the multiple_converge / idempotency checks enabled?
```yaml

provisioner:
  name: chef_solo
  enforce_idempotency: true
  multiple_converge: 2
  deprecations_as_errors: true
```

File: templates/default/inspec_default_test.rb.erb [hunk 2]
```diff
@@ -1,4 +1,4 @@
 # InSpec test for recipe <%= cookbook_name %>::<%= recipe_name %>
 
 # The InSpec reference, with examples and extensive documentation, can be
-# found at http://inspec.io/docs/reference/resources/
+# found at https://www.inspec.io/docs/reference/resources/
```
Comment (detjensrobert) [inline templates/default/inspec_default_test.rb.erb:4]: Inspec docs have moved:
```suggestion
# found at https://docs.chef.io/inspec/resources/
```

Comment (ramereth) [reply templates/default/inspec_default_test.rb.erb:4]: You should probably make an update in the the upstream chef-cli repo then as I copied this directly from there :laughing:

**Pull request merged**